engine/engine.go

```diff
@@ -699,7 +715,9 @@ func runAvailableSteps(dbp zesty.DBProvider, modifiedSteps map[string]bool, res
 				if s.State != step.StateAfterrunError {
 					res.SetStepState(s.Name, step.StateRunning)
 					step.PreRun(s, res.Values, resolutionStateSetter(res, preRunModifiedSteps), executedSteps)
-					commit(dbp, res, nil)
+					if err := commit(dbp, res, nil); err != nil {
+						return 0, err
```

Comment:
Problem in my current implementation is that exiting the `runAvailableSteps` here, will close the main loop, while some step might still be running.

Not sure how we should handle this.